### PR TITLE
Rename sha3 too keccak256

### PR DESCRIPTION
**Pull request by adauto82**
This is just a renaming of some classes and methods to reflect the hash function we are actually using and avoid confusion.

This was asked by @juli via programming channel in slack

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 705 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 19 critical
* ![MAJOR][MAJOR] 419 major
* ![MINOR][MINOR] 54 minor
* ![INFO][INFO] 212 info

#### Top 10 issues

1. ![BLOCKER][BLOCKER] [Web3Impl.java#L504](https://github.com/rsksmart/rskj/blob/32662256df504a5280d2a93130622882c0cc9488/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L504): "blockchain" is a method parameter, and should not be used for synchronization. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2445)
1. ![CRITICAL][CRITICAL] [Wallet.java#L217](https://github.com/rsksmart/rskj/blob/32662256df504a5280d2a93130622882c0cc9488/rskj-core/src/main/java/co/rsk/core/Wallet.java#L217): Object deserialization is used in co.rsk.core.Wallet.decryptAES(byte[], byte[]) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3AOBJECT_DESERIALIZATION)
1. ![CRITICAL][CRITICAL] [PendingStateImpl.java#L48](https://github.com/rsksmart/rskj/blob/32662256df504a5280d2a93130622882c0cc9488/rskj-core/src/main/java/co/rsk/core/bc/PendingStateImpl.java#L48): Unused import - org.ethereum.crypto.HashUtil.keccak256. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [RepositoryImpl.java#L48](https://github.com/rsksmart/rskj/blob/32662256df504a5280d2a93130622882c0cc9488/rskj-core/src/main/java/co/rsk/db/RepositoryImpl.java#L48): Unused import - org.ethereum.crypto.Keccak256Helper.keccak256. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [MinerServerImpl.java#L344](https://github.com/rsksmart/rskj/blob/32662256df504a5280d2a93130622882c0cc9488/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L344): Make the enclosing method "static" or remove this set. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2696)
1. ![CRITICAL][CRITICAL] [MinerServerImpl.java#L348](https://github.com/rsksmart/rskj/blob/32662256df504a5280d2a93130622882c0cc9488/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L348): Make the enclosing method "static" or remove this set. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2696)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L85](https://github.com/rsksmart/rskj/blob/32662256df504a5280d2a93130622882c0cc9488/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L85): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L92](https://github.com/rsksmart/rskj/blob/32662256df504a5280d2a93130622882c0cc9488/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L92): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L100](https://github.com/rsksmart/rskj/blob/32662256df504a5280d2a93130622882c0cc9488/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L100): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L112](https://github.com/rsksmart/rskj/blob/32662256df504a5280d2a93130622882c0cc9488/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L112): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/test/java/co/rsk/peg/StateForFederatorTest.java [hunk 1]
```diff
@@ -44,23 +44,23 @@
 
     @Test
     public void serialize() {
-        Sha3Hash sha3Hash1 = new Sha3Hash(SHA3_1);
-        Sha3Hash sha3Hash2 = new Sha3Hash(SHA3_2);
-        Sha3Hash sha3Hash3 = new Sha3Hash(SHA3_3);
-        Sha3Hash sha3Hash4 = new Sha3Hash(SHA3_4);
+        Keccak256 keccak2561 = new Keccak256(SHA3_1);
```
Comment (tinchou) [inline rskj-core/src/test/java/co/rsk/peg/StateForFederatorTest.java]: these are not very readable now. maybe `hash1`...`hash4`?

File: rskj-core/src/main/java/co/rsk/core/bc/PendingStateImpl.java [hunk 2]
```diff
@@ -44,15 +45,15 @@
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 
-import static org.ethereum.crypto.HashUtil.sha3;
+import static org.ethereum.crypto.HashUtil.keccak256;
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/core/bc/PendingStateImpl.java]: Please fix unused imports!
You also replace some static usages with fully qualified usages later on, maybe you want to make that coherent.

**Comment by rskops** [comment]
SonarQube analysis reported 706 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 19 critical
* ![MAJOR][MAJOR] 419 major
* ![MINOR][MINOR] 54 minor
* ![INFO][INFO] 213 info

#### Top 10 issues

1. ![BLOCKER][BLOCKER] [Web3Impl.java#L504](https://github.com/rsksmart/rskj/blob/b9be35ede483dcd0ed28c09e04f408a45c1fdfea/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L504): "blockchain" is a method parameter, and should not be used for synchronization. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2445)
1. ![CRITICAL][CRITICAL] [Wallet.java#L217](https://github.com/rsksmart/rskj/blob/b9be35ede483dcd0ed28c09e04f408a45c1fdfea/rskj-core/src/main/java/co/rsk/core/Wallet.java#L217): Object deserialization is used in co.rsk.core.Wallet.decryptAES(byte[], byte[]) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3AOBJECT_DESERIALIZATION)
1. ![CRITICAL][CRITICAL] [PendingStateImpl.java#L48](https://github.com/rsksmart/rskj/blob/b9be35ede483dcd0ed28c09e04f408a45c1fdfea/rskj-core/src/main/java/co/rsk/core/bc/PendingStateImpl.java#L48): Unused import - org.ethereum.crypto.HashUtil.keccak256. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [RepositoryImpl.java#L48](https://github.com/rsksmart/rskj/blob/b9be35ede483dcd0ed28c09e04f408a45c1fdfea/rskj-core/src/main/java/co/rsk/db/RepositoryImpl.java#L48): Unused import - org.ethereum.crypto.Keccak256Helper.keccak256. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [MinerServerImpl.java#L344](https://github.com/rsksmart/rskj/blob/b9be35ede483dcd0ed28c09e04f408a45c1fdfea/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L344): Make the enclosing method "static" or remove this set. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2696)
1. ![CRITICAL][CRITICAL] [MinerServerImpl.java#L348](https://github.com/rsksmart/rskj/blob/b9be35ede483dcd0ed28c09e04f408a45c1fdfea/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L348): Make the enclosing method "static" or remove this set. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2696)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L85](https://github.com/rsksmart/rskj/blob/b9be35ede483dcd0ed28c09e04f408a45c1fdfea/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L85): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L92](https://github.com/rsksmart/rskj/blob/b9be35ede483dcd0ed28c09e04f408a45c1fdfea/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L92): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L100](https://github.com/rsksmart/rskj/blob/b9be35ede483dcd0ed28c09e04f408a45c1fdfea/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L100): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L112](https://github.com/rsksmart/rskj/blob/b9be35ede483dcd0ed28c09e04f408a45c1fdfea/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L112): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/core/bc/PendingStateImpl.java (hunk 2 above)
Comment (tinchou) [reply rskj-core/src/main/java/co/rsk/core/bc/PendingStateImpl.java]: either all `HashUtil.keccak256` or all `keccak256`, I'm fine with both.

**Comment by rskops** [comment]
SonarQube analysis reported 706 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 19 critical
* ![MAJOR][MAJOR] 419 major
* ![MINOR][MINOR] 54 minor
* ![INFO][INFO] 213 info

#### Top 10 issues

1. ![BLOCKER][BLOCKER] [Web3Impl.java#L504](https://github.com/rsksmart/rskj/blob/ba35aba1afc23a538e375b7c46fbe1c8bd9ea429/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L504): "blockchain" is a method parameter, and should not be used for synchronization. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2445)
1. ![CRITICAL][CRITICAL] [Wallet.java#L217](https://github.com/rsksmart/rskj/blob/ba35aba1afc23a538e375b7c46fbe1c8bd9ea429/rskj-core/src/main/java/co/rsk/core/Wallet.java#L217): Object deserialization is used in co.rsk.core.Wallet.decryptAES(byte[], byte[]) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3AOBJECT_DESERIALIZATION)
1. ![CRITICAL][CRITICAL] [PendingStateImpl.java#L48](https://github.com/rsksmart/rskj/blob/ba35aba1afc23a538e375b7c46fbe1c8bd9ea429/rskj-core/src/main/java/co/rsk/core/bc/PendingStateImpl.java#L48): Unused import - org.ethereum.crypto.HashUtil.keccak256. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [RepositoryImpl.java#L48](https://github.com/rsksmart/rskj/blob/ba35aba1afc23a538e375b7c46fbe1c8bd9ea429/rskj-core/src/main/java/co/rsk/db/RepositoryImpl.java#L48): Unused import - org.ethereum.crypto.Keccak256Helper.keccak256. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [MinerServerImpl.java#L344](https://github.com/rsksmart/rskj/blob/ba35aba1afc23a538e375b7c46fbe1c8bd9ea429/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L344): Make the enclosing method "static" or remove this set. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2696)
1. ![CRITICAL][CRITICAL] [MinerServerImpl.java#L348](https://github.com/rsksmart/rskj/blob/ba35aba1afc23a538e375b7c46fbe1c8bd9ea429/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L348): Make the enclosing method "static" or remove this set. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2696)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L85](https://github.com/rsksmart/rskj/blob/ba35aba1afc23a538e375b7c46fbe1c8bd9ea429/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L85): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L92](https://github.com/rsksmart/rskj/blob/ba35aba1afc23a538e375b7c46fbe1c8bd9ea429/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L92): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L100](https://github.com/rsksmart/rskj/blob/ba35aba1afc23a538e375b7c46fbe1c8bd9ea429/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L100): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L112](https://github.com/rsksmart/rskj/blob/ba35aba1afc23a538e375b7c46fbe1c8bd9ea429/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L112): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**